docs/guide_slash_and_interactions.rst

```diff
@@ -0,0 +1,292 @@
+
+---------------
+Getting Started
+---------------
+
+To start off, we will have to import some additional modules to our cog file.
+We will be using the :class:`redbot.core.app_commands` module to create our slash commands.
+Once we have imported the module, we can start creating our slash commands in our cog class.
+For this example we will use a basic hello world command.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        async def hello(self, interaction: discord.Interaction):
+            await interaction.response.send_message("Hello World!", ephemeral=True)
+
+Go ahead and load your cog. Once it is loaded, we will have to enable and sync our slash commands.
+We can do this by using the :ref:`[p]slash<core-command-slash>` command to manage our slash commands.
+Once you have registered your slash commands, you can test them out by typing ``/hello`` in your server.
+
+----------------------------
+Slash Commands and Arguments
+----------------------------
+
+There is a lot of flexibility when it comes to slash commands.
+Below we will go over some of the different stuff you can do with slash commands.
+
+Decorators
+----------
+Just like with text commands, we can use decorators to modify the behaviour of our slash commands.
+For example, we can use the :func:`app_commands.guild_only` decorator to make our slash command only work in guilds.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.guild_only()
+        async def hello(self, interaction: discord.Interaction):
+            await interaction.response.send_message("Hello World!", ephemeral=True)
+
+One of the more useful decorators is the :func:`app.commands.choices` decorator.
+This decorator allows us to specify a list of choices for a specific argument.
+This is useful for arguments that have a limited number of options.
+For example, we can use this to create a command that allows us to choose between two different colors.
+
+.. code-block:: python
+
+    from enum import Enum
+    from redbot.core import commands, app_commands
+
+    class Color(Enum):
+        Red = 1
+        Blue = 2
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.describe(color="The color you want to choose")
+        async def color(self, interaction: discord.Interaction, color: Color):
+            await interaction.response.send_message(f"Your color is {str(color.name)}", ephemeral=True)
+
```

Comment:
```suggestion

.. code-block:: python

    from redbot.core import commands, app_commands

    class MyCog(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        @app_commands.command()
        @app_commands.describe(color="The color you want to choose")
        @app_commands.choices(color=[
             app_commands.Choice(name="Red', value="red"),
             app_commands.Choice(name="Blue', value="blue"),
        ])
        async def color(self, interaction: discord.Interaction, color: Color):
            await interaction.response.send_message(f"Your color is {str(color)}", ephemeral=True)

The user will be shown the ``name`` of the choice, and the argument will be passed the
``value`` associated with that choice. This allows user-facing names to be prettier than
what is actually processed by the command.

Alternatively, ``Literal`` can be used if the argument does not need a different
user-facing label.

.. code-block:: python

    from redbot.core import commands, app_commands
    from typing import Literal

    class MyCog(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        @app_commands.command()
        @app_commands.describe(color="The color you want to choose")
        async def color(self, interaction: discord.Interaction, color: Literal["Red", "Blue"]):
            await interaction.response.send_message(f"Your color is {str(color)}", ephemeral=True)

Finally, an ``Enum`` subclass can be used to specify choices. When done this way, the
resulting parameter will be an instance of that enum, rather than the ``value``.

.. code-block:: python

    from enum import Enum
    from redbot.core import commands, app_commands

    class Color(Enum):
        Red = "red"
        Blue = "blue"

    class MyCog(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        @app_commands.command()
        @app_commands.describe(color="The color you want to choose")
        async def color(self, interaction: discord.Interaction, color: Color):
            await interaction.response.send_message(f"Your color is {str(color.value)}", ephemeral=True)

```